x-pack/packages/security-solution/distribution_bar/src/distribution_bar.tsx

```diff
@@ -20,56 +21,163 @@ export interface EmptyBarProps {
+        }
+      `,
+      // last tooltip should be displayed initially even without hover
+      lastTooltip: css`
+        &:last-child > div {
+          opacity: 1;
+        }
+        &:last-child::after {
+          opacity: 1;
+        }
+        &:hover ~ div:last-child > div {
+          opacity: 0;
+        }
+        &:hover ~ div:last-child::after {
+          opacity: 0;
+        }
+      `,
+    },
+    tooltip: css`
+      opacity: 0;
+      position: absolute;
+      width: 100%;
+      height: calc(
+        ${euiTheme.base + 2}px + 14px + 7px
+      ); // 2px border of the badge + 14px height of the tick + 7px height of the bar
+      text-align: right;
+      top: calc(
+        -${euiTheme.base + 2}px - 14px
+      ); // 2px border of the badge + 14px height of the tick
+      right: 0;
+
+      &:hover {
+        opacity: 1;
+        top: calc(
+          -${euiTheme.base + 2}px - 13px
+        ); // 2px border of the badge + 14px height of the tick - 1px to accomodate for height of hovered bar
+        transition: all 0.3s ease;
+      }
+    `,
+    tooltipBadgeLeft: css`
+      border-bottom-right-radius: 0;
+      border-top-right-radius: 0;
+    `,
+    tooltipBadgeRight: css`
+      border-left: none;
+      border-bottom-left-radius: 0;
+      border-top-left-radius: 0;
+    `,
+  };
 };
 
 const EmptyBar: React.FC<EmptyBarProps> = ({ 'data-test-subj': dataTestSubj }) => {
-  const { euiTheme } = useEuiTheme();
+  const styles = useStyles();
 
-  const emptyBarStyle = [
-    styles.base,
-    css`
-      background-color: ${euiTheme.colors.lightShade};
-      flex: 1;
-    `,
-  ];
+  const emptyBarStyle = [styles.part.base, styles.part.empty];
 
-  return <span css={emptyBarStyle} data-test-subj={`${dataTestSubj}__emptyBar`} />;
+  return <div css={emptyBarStyle} data-test-subj={`${dataTestSubj}`} />;
 };
 
+// TODO: fix tooltip direction if not enough space;
 /**
  * Security Solution DistributionBar component.
  * Shows visual representation of distribution of stats, such as alerts by criticality or misconfiguration findings by evaluation result.
  */
 export const DistributionBar: React.FC<DistributionBarProps> = React.memo(function DistributionBar(
   props
 ) {
-  const { euiTheme } = useEuiTheme();
+  const styles = useStyles();
   const { stats, 'data-test-subj': dataTestSubj } = props;
   const parts = stats.map((stat) => {
     const partStyle = [
-      styles.base,
+      styles.part.base,
+      styles.part.tick,
+      styles.part.hover,
+      styles.part.lastTooltip,
       css`
         background-color: ${stat.color};
         flex: ${stat.count};
       `,
     ];
+    const prettyNumber = numeral(stat.count).format('0,0a');
 
-    return <span key={stat.key} css={partStyle} />;
+    return (
+      <div key={stat.key} css={partStyle} data-test-subj={`${dataTestSubj}__part`}>
+        <div css={styles.tooltip}>
+          <EuiFlexGroup gutterSize={'none'} justifyContent={'flexEnd'} wrap={false}>
```

Comment:
It seems it gets awkward on smaller screens, I guess it's because of the [responsiveness](https://eui.elastic.co/#/layout/flex%23responsive-flex-groups) of the EuiFlexGroup component

![image](https://github.com/user-attachments/assets/ec0f35cf-d099-48d8-b8ab-825befc2aa88)


it just a responsiveness issue, it works well on containers with smaller width on the desktop

Reply:
thanks, didn't know about that! fixed